library/hls/src/main/java/com/google/android/exoplayer2/source/hls/playlist/DefaultHlsPlaylistTracker.java

```diff
@@ -196,6 +196,15 @@ public HlsMediaPlaylist getPlaylistSnapshot(Uri url, boolean isForPlayback) {
     return snapshot;
   }
 
+  @Nullable
+  @Override
+  public HlsMediaPlaylist getFreshPrimaryPlaylist(Uri url) {
+    MediaPlaylistBundle playlistBundle = playlistBundles.get(url);
+    boolean isPrimary = url.equals(primaryMediaPlaylistUrl);
+    boolean isFresh = playlistBundle.isSnapshotFresh();
+    return isPrimary && isFresh ? playlistBundle.getPlaylistSnapshot() : null;
+  }
+
```

Comment:
Only the current primary HlsMediaPlaylist is up to date with the Period/Timeline used by ExoPlayerImplInternal, and only if it has been loaded (and thus reported the `MediaSource`.